pkg/scalers/azure_eventhub_scaler.go

```diff
@@ -156,37 +169,70 @@ func parseAzureEventHubMetadata(config *ScalerConfig) (*eventHubMetadata, error)
 	}
 	serviceBusEndpointSuffix, err := azure.ParseEnvironmentProperty(config.TriggerMetadata, azure.DefaultEndpointSuffixKey, serviceBusEndpointSuffixProvider)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	meta.eventHubInfo.ServiceBusEndpointSuffix = serviceBusEndpointSuffix
 
 	activeDirectoryEndpoint, err := azure.ParseActiveDirectoryEndpoint(config.TriggerMetadata)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	meta.eventHubInfo.ActiveDirectoryEndpoint = activeDirectoryEndpoint
 
+	meta.scalerIndex = config.ScalerIndex
+
+	return nil
+}
+
+func parseAzureEventHubAuthenticationMetadata(logger logr.Logger, config *ScalerConfig, meta *eventHubMetadata) error {
 	meta.eventHubInfo.PodIdentity = config.PodIdentity
+
 	switch config.PodIdentity.Provider {
 	case "", v1alpha1.PodIdentityProviderNone:
+		if len(meta.eventHubInfo.StorageConnection) == 0 {
+			return fmt.Errorf("no storage connection string given")
+		}
+
 		if config.AuthParams["connection"] != "" {
 			meta.eventHubInfo.EventHubConnection = config.AuthParams["connection"]
 		} else if config.TriggerMetadata["connectionFromEnv"] != "" {
 			meta.eventHubInfo.EventHubConnection = config.ResolvedEnv[config.TriggerMetadata["connectionFromEnv"]]
 		}
 
 		if len(meta.eventHubInfo.EventHubConnection) == 0 {
-			return nil, fmt.Errorf("no event hub connection string given")
+			return fmt.Errorf("no event hub connection string given")
 		}
 	case v1alpha1.PodIdentityProviderAzure, v1alpha1.PodIdentityProviderAzureWorkload:
+		meta.eventHubInfo.StorageAccountName = ""
+		if val, ok := config.TriggerMetadata["storageAccountName"]; ok {
+			meta.eventHubInfo.StorageAccountName = val
+		} else {
+			logger.Info("no 'storageAccountName' provided to enable identity based authentication to Blob Storage. Attempting to use connection string instead")
+		}
+
+		if len(meta.eventHubInfo.StorageAccountName) != 0 {
+			storageEndpointSuffixProvider := func(env az.Environment) (string, error) {
+				return env.StorageEndpointSuffix, nil
+			}
+			storageEndpointSuffix, err := azure.ParseEnvironmentProperty(config.TriggerMetadata, azure.DefaultStorageSuffixKey, storageEndpointSuffixProvider)
+			if err != nil {
+				return err
+			}
+			meta.eventHubInfo.BlobStorageEndpoint = "blob." + storageEndpointSuffix
+		}
+
+		if len(meta.eventHubInfo.StorageConnection) == 0 && len(meta.eventHubInfo.StorageAccountName) == 0 {
+			return fmt.Errorf("no storage connection string or storage account name for pod identity based authentication given")
+		}
+
```

Comment:
I'm not sure if we should do a fallback with configuration parameters, I mean, if users set manage identities and they don't provide all the required data, IMO we should raise a clear error, not fallback to another auth mechanism because the users have set specifically they want managed identities.
WDYT @kedacore/keda-core-contributors ?

Reply:
I think the problem is, currently if someone is using Event Hub scaler with pod identity they would have manually provided a connection string. 

If we throw an error here, if a user upgrades their KEDA installation, then KEDA would throw errors on their cluster because the identity being used for event hub might not have access to the storage account.

Reply:
okey... In that case, the used identity should have all the needed access, right?. 
I don't get the advantage of having pod identity if we still need the connection string, it sounds weird to me. 
I mean, why would I want to use a manage identity which has access to event hub but not to the storage itself?
I get the point that with connection strings, we need both because we can't have only 1 for 2 different services, but with manage identities I thought we can use the same for both.
If you think this is legit, ignore my comments, you are the expert on this xD I said only because it's weird to me

Reply:
No, you are right, it doesn't make sense to have the identity have access to the event hub but not storage. But some users might be having such a case due to Event Hub scaler not supporting managed identity for checkpointing storage check. 

Ideally, we should not support connection strings when using managed identity. But we need to make sure this change isn't breaking.

Reply:
As @v-shenoy points out, the situation of supporting both identity and connection string isn't ideal but necessary not to break any existing deployments by accident. You could consider deprecating the configuration and starting to throw a warning in a future release, then remove the scenario entirely in favor of a pod identity-based approach. That would be a break in functionality as indicated and would have to follow whichever process keda favors for such a change, best not attempted here while introducing the new feature.

Reply:
Can this comment be resolved?

Reply:
Resolving the comment due to no follow up and the points mentioned above. @JorTurFer, please reopen if you feel otherwise.